Assets/Scripts/TES/MorrowindEngine.cs

```diff
@@ -285,33 +286,33 @@ public void Update()
 		}
 		public void CastInteractRay()
 		{
-			interactTextObj.SetActive(false);
-
 			// Cast a ray to see what the camera is looking at.
-			var ray = new Ray(playerCameraObj.transform.position, playerCameraObj.transform.forward);
+			Ray ray = new Ray(playerCameraObj.transform.position, playerCameraObj.transform.forward);
 
-			var raycastHitCount = Physics.RaycastNonAlloc(ray, interactRaycastHitBuffer, maxInteractDistance);
+			int raycastHitCount = Physics.RaycastNonAlloc(ray, interactRaycastHitBuffer, maxInteractDistance);
 
 			for(int i = 0; i < raycastHitCount; i++)
 			{
-				var hitInfo = interactRaycastHitBuffer[i];
+				RaycastHit hitInfo = interactRaycastHitBuffer[i];
 
 				// Find the door associated with the hit collider.
 				GameObject doorObj = GameObjectUtils.FindObjectWithTagUpHeirarchy(hitInfo.collider.gameObject, "Door");
 
 				if(doorObj != null)
 				{
-					var doorComponent = doorObj.GetComponent<DoorComponent>();
+					DoorComponent doorComponent = doorObj.GetComponent<DoorComponent>();
 
-					interactTextObj.SetActive(true);
+					if ( !interactTextObj.activeSelf ) interactTextObj.SetActive( true );
```

Comment:
Noted! the var thing was most likely me trying to follow the code better. having never seen the classes and types that this engine uses, I'm still learning the structure. I'll try to use var in the same way you do, and I'll change my formatting settings in Visual Studio to match yours as closely as I can.